packages/discord.js-utilities/src/lib/PaginatedMessages/PaginatedMessage.ts

```diff
@@ -28,14 +28,51 @@ import { isGuildBasedChannel } from '../type-guards';
  *
  * @example
  * ```typescript
- * const handler = new PaginatedMessage();
+ * const myPaginatedMessage = new PaginatedMessage();
+ * // Once you have an instance of PaginatedMessage you can call various methods on it to add pages to it.
+ * // For more details see each methods' documentation.
+ *
+ * myPaginatedMessage.addPageEmbed((embed) => {
+ *		embed
+ *			.setColor('#FF0000')
+ *			.setDescription('example description');
+ *
+ *		return embed;
+ * });
+ *
+ * myPaginatedMessage.addPageBuilder((builder) => {
+ *		const embed = new MessageEmbed()
+ *			.setColor('#FF0000')
+ *			.setDescription('example description');
+ *
+ *		return builder
+ *			.setContent('example content')
+ *			.setEmbed(embed);
+ * });
+ *
+ * myPaginatedMessage.addPageContent('Example');
+ *
+ * myPaginatedMessage.run(message)
  * ```
  *
+ * @remark You can also provide a MessageEmbed template. This will be applied to every page.
+ * If a page itself has an embed then the two will be merged into 1, with the data of
+ * the page's own embed taking higher priority over that of the template.
+ *
+ * Furthermore, if the template has a footer then it will be applied _after_ the page index part of the footer
+ * and a space will be added between the index and your provided footer. For example when setting `- Powered by Sapphire Framework`
+ * the resulting footer will be `1/2 - Powered by Sapphire Framework`
  * @example
  * ```typescript
- * // To utilize actions you can use the IPaginatedMessageAction by implementing it into a class.
- * // PaginatedMessage requires you to have the class initialized using `new`.
+ * const myPaginatedMessage = new PaginatedMessage({
+ * 	template: new MessageEmbed().setColor('#FF0000').setFooter('- powered by sapphire framework')
+ * });
+ * ```
  *
+ * @remark To utilize actions you can use the IPaginatedMessageAction by implementing it into a class.
```

Comment:
```suggestion
 * @remark To utilize actions you can implement IPaginatedMessageAction into a class.
```
Typical lecture about passive voice and conciseness goes here